### PR TITLE
Added validation for quantity of pills to be entered on POC Encounter Form

### PR DESCRIPTION
**Pull request by dbaluku**
https://app.asana.com/0/1133167201254915/1177037776863329/f

### CODE REVIEW

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 1]
```diff
@@ -724,9 +731,40 @@
...
+                     getField('no-of-art-pills.error').html('No. of pills required').show();
+                    return false;
+                }
+
+                if (getValue('isoniazid-trigger.value') != "" &amp;&amp; getValue('isoniazid-pills.value') == "") {
+                    getField('isoniazid-pills.error').html('No. of pills required').show();
+                    return false;
+                }
+
+                if (getValue('b6-trigger.value') != "" &amp;&amp; getValue('b6-pills.value') == "") {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku What is b6? can you use a more specific ID

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 2]
```diff
@@ -724,9 +731,40 @@
...
+                    getField('b6-pills.error').html('No. of pills required').show();
+                    return false;
+                }
+
+                if (getValue('99413-target.value') != "" &amp;&amp; getValue('160856.value') == "") {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1134]: @dbaluku add a comment here what is 99413?

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 3]
```diff
@@ -724,9 +731,40 @@
...
+                    getField('160856.error').html('No. of pills required').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku the error message field used is different from the one checked for a value, can you describe what these pills are

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 4]
```diff
@@ -724,9 +731,40 @@
...
+                    return false;
+                }
+
+                if (getValue('fluconazole-trigger.value') != "" &amp;&amp; getValue('fluconazole-pills.value') == "") {
+                    getField('fluconazole-pills.error').html('No. of pills required').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku message 'No. of fluconazole pills required'

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 5]
```diff
@@ -724,9 +731,40 @@
...
+                    return false;
+                }
+
+                if (getValue('other-meds-trigger.value') != "" &amp;&amp; getValue('other-meds-quantity.value') == "") {
+                    getField('other-meds-quantity.error').html('Quantity required').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku message 'Quantity of other medications is required' - can you get the name of the selected medication to make this more user friendly

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 6]
```diff
@@ -724,9 +731,40 @@
...
                     return false;
                 }
+
+                if (getValue('art-regimen.value') != "" &amp;&amp; getValue('no-of-art-pills.value') == "") {
+                     getField('no-of-art-pills.error').html('No. of pills required').show();
+                    return false;
+                }
+
+                if (getValue('isoniazid-trigger.value') != "" &amp;&amp; getValue('isoniazid-pills.value') == "") {
+                    getField('isoniazid-pills.error').html('No. of pills required').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku error message - 'No of isoniazid pills required'

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 7]
```diff
@@ -724,9 +731,40 @@
                     jq().toastmessage('showErrorToast', " Patient Not Stable " + instabilityCause + ". \n Only Stable Patients can be enrolled in " + jq("#165143").find("select").find(":selected").text());
                     return false;
                 }
+
+                if (getValue('art-regimen.value') != "" &amp;&amp; getValue('no-of-art-pills.value') == "") {
+                     getField('no-of-art-pills.error').html('No. of pills required').show();
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku Error message - No of art-regimen pills required. Put the selected ART regimen as part of the error message

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml [hunk 8]
```diff
@@ -2775,13 +2813,13 @@
...
                                                         <br/>
-                                                        <obs conceptId="1282" answerConceptSetIds="160021"
+                                                        <obs id="b6-trigger" conceptId="1282" answerConceptSetIds="160021"
                                                              style="radio"
                                                              answerConceptIds="82912" answerLabels="Vitamin B6"/>
                                                     </div>
                                                     <div class="col-md-4">
                                                         <label>No. of Pills</label>
-                                                        <obs conceptId="160856"/>
+                                                        <obs id="b6-pills" conceptId="160856"/>
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @dbaluku rename this field to vitamin-b6 (for clarity - had confused me earlier)

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 1 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: @ssmusoke b6 is a drug on the ecounter page

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 2 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml:1134]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 3 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 4 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 5 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 6 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 7 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml (hunk 8 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/HMIS-HIV-003-HivCareArtCard-ClinicalAssessmentPage.xml]: Done

**Pull request merged**